File: src/main/java/com/matag/game/player/playerInfo/PlayerInfoRetriever.java
package com.matag.game.player.playerInfo;

import org.springframework.stereotype.Component;

import com.matag.adminentities.PlayerInfo;
import com.matag.game.adminclient.AdminClient;
import com.matag.game.security.SecurityToken;

import lombok.AllArgsConstructor;

@AllArgsConstructor
@Component
public class PlayerInfoRetriever {
  private final AdminClient adminClient;

  public PlayerInfo retrieve(SecurityToken token) {
    return adminClient.getPlayerInfo(token);
  }
}

File: src/main/kotlin/com/matag/game/player/playerInfo/PlayerInfoRetriever.kt
package com.matag.game.player.playerInfo

import com.matag.adminentities.PlayerInfo
import com.matag.game.adminclient.AdminClient
import com.matag.game.security.SecurityToken
import org.springframework.stereotype.Component

@Component
class PlayerInfoRetriever(
    private val adminClient: AdminClient
) {
    fun retrieve(token: SecurityToken?): PlayerInfo? {
        return adminClient.getPlayerInfo(token)
    }
}
